src/Features/VisualBasic/Portable/CodeRefactorings/InlineTemporary/InlineTemporaryCodeRefactoringProvider.vb

```diff
@@ -280,12 +280,89 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.CodeRefactorings.InlineTemporary
             Return localDeclaration.Parent
         End Function
 
+        Private Function GetNonWhitespaceTrivia(trivia As IEnumerable(Of SyntaxTrivia)) As IEnumerable(Of SyntaxTrivia)
+            Dim newLeadingTrivia = trivia _
+                .Reverse _
+                .SkipWhile(Function(t) t.IsKind(SyntaxKind.WhitespaceTrivia, SyntaxKind.EndOfLineTrivia)) _
+                .Reverse
+
+            ' Ensure that we leave a line break if trivia with a comment.
+            If newLeadingTrivia.Any() AndAlso newLeadingTrivia.Last().IsKind(SyntaxKind.CommentTrivia) Then
+                newLeadingTrivia = newLeadingTrivia.Concat(SyntaxFactory.CarriageReturnLineFeed)
+            End If
+            Return newLeadingTrivia
+        End Function
+
         Private Function GetUpdatedDeclaration(modifiedIdentifier As ModifiedIdentifierSyntax) As LocalDeclarationStatementSyntax
             Dim variableDeclarator = DirectCast(modifiedIdentifier.Parent, VariableDeclaratorSyntax)
             Dim localDeclaration = DirectCast(variableDeclarator.Parent, LocalDeclarationStatementSyntax)
 
             If localDeclaration.Declarators.Count > 1 And variableDeclarator.Names.Count = 1 Then
-                Return localDeclaration.RemoveNode(variableDeclarator, SyntaxRemoveOptions.KeepEndOfLine)
+                Dim newLocalDeclaration = localDeclaration.RemoveNode(variableDeclarator, SyntaxRemoveOptions.KeepNoTrivia)
+                Dim index = localDeclaration.Declarators.IndexOf(variableDeclarator)
+
+                If index = 0 Then
```

Comment:
This all seems super complex.  Why not just grab the leading comment trivia and put that on the resulttant node if possible.

Do we really need all these cases?